### ISSUE TITLE
Turn RFCs into developer docs

### ISSUE THREAD

**Issue by meiqimichelle** [ipfs/ipfs-cluster-website#78]
We have: https://cluster.ipfs.io/documentation/developer/rfcs/
And then we have: https://cluster.ipfs.io/documentation/developer/rfcs/dag-sharding-rfc/
And probably some others floating around. Where should they live in GH and/or the website?

**Comment by hsanjuan** [comment]
They should probably be removed. RFCs should turn into developer docs that actually explain how things were done in the end.

RFCs should be good to discuss an idea but they might lead to confusion if they're very out of date, on the website mixed with other docs.

**Comment by kishansagathiya** [comment]
It would be nice to have a link somewhere for RFC PRs(they have more info that RFCs itself). For extra curious soul, they provide a way to see why some development decision was made.

**Comment by lanzafame** [comment]
Braindump of some of things I have been thinking on this topic recently, after KK and the recent Follower/CRDT RFC confusion. 

Personally, I think an RFC should define what the interface should do and not how it is implemented. An example would be an RFC defining what part the `PinTracker` plays in the overall Cluster operation. An RFC that defines that shouldn't get out of date and if it does, there should be an RFC that supersedes it. A more high-level RFC would be the one that defines what a `Follower` is, it would list what components it interacts with, the fact that it doesn't expose a new RPC API, etc.

On the developer docs side, they should cover how to integrate with cluster and what some of the important technical design decisions were made in the implementation that may affect their integration, i.e. what are the consequences of integrating with the default `Client` vs the `LoadBalancingClient` etc. 

To summarise, I agree that we shouldn't turn RFCs into dev docs as they are different things and we shouldn't have RFCs on the website. I do however think that we should have RFCs the get merged instead of our current weird RFC PR practices, whether they get merged into ipfs/ipfs-cluster or into a new repo like ipfs-cluster/rfcs, I am not sure but they should exist in a merged state somewhere.
